### PR TITLE
images: Install thin-provisioning-tools from Debian unstable on debian-testing

### LINKED ISSUE #5295: Image refresh for debian-testing
Image refresh for debian-testing
 * [ ] Fix https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=1041089 and get thin-provisioning-tools back into testing
 * [ ] FAIL: image-refresh debian-testing

### PR DESCRIPTION
**Pull request by martinpitt**
thin-provisioning-tools fell out of testing three weeks ago, which broke our debian-testing image build. The fix is underway, but will still take some time to land.

To avoid stalling our debian-testing coverage further, install it from unstable for the time being.

Fixes #5295

 * [x] image-refresh debian-testing
 * [x] cockpit tests get fixed by https://github.com/cockpit-project/cockpit/pull/19462
 * [x] c-podman tests get fixed by https://github.com/cockpit-project/cockpit-podman/pull/1440

### CODE REVIEW

**Comment by cockpituous** [comment]
Failed. Log: https://cockpit-logs.us-east-1.linodeobjects.com/image-refresh-logs/debian-testing-20231009-074141.log

**Comment by martinpitt** [comment]
Meh, seems pcp now hangs during shutdown. Will investigate, but not right now.

**Comment by cockpituous** [comment]
image-refresh debian-testing done: https://github.com/cockpit-project/bots/commits/image-refresh-debian-testing-20231009-155849

**Comment by martinpitt** [comment]
Ouch, lots of TODOs here. At least the cockpit test looks the same as arch and f39, but podman and machines have new failures.

**Comment by martinpitt** [comment]
This [AppArmor violation](https://cockpit-logs.us-east-1.linodeobjects.com/pull-5358-20231009-162701-bc35b231-debian-testing-cockpit-project-cockpit-machines/log.html#55) seems to be a flake. Retrying to confirm -- I'll report it anyway, just want to know whether it broke specifically with this update.

File: images/scripts/debian.setup [hunk 1]
```diff
@@ -116,6 +116,18 @@ if [ "$1" = "debian-testing" ]; then
...
+    echo 'deb http://deb.debian.org/debian unstable main' >> /etc/apt/sources.list
+    cat <<EOF > /etc/apt/preferences.d/unstable
+Package: *
+Pin: release a=unstable
+Pin-Priority: -1
+
+Package: thin-provisioning-tools
+Pin: release a=unstable
+Pin-Priority: 10
+EOF
```
Comment (jelly) [inline images/scripts/debian.setup:130]: Does this mean an `apt upgrade` will also update to unstable? Not that it matters too much as this is hopefully temporary but good to know.

**Review by jelly — APPROVED** [review]

**Pull request merged**